core/txbuilder/constraint.go

```diff
@@ -41,7 +41,8 @@ func (t timeConstraint) code() []byte {
 		}
 		builder.AddOp(vm.OP_MAXTIME).AddInt64(int64(t.maxTimeMS)).AddOp(vm.OP_LESSTHANOREQUAL)
 	}
-	return builder.Program
+	prog, _ := builder.Build()
```

Comment:
This change seems to encourage ignoring errors for programs known to be "safe". I have a slight nag in my head that now that the signature has changed, the implementation can change, and outside expectations won't be met. I can't think of any reasonable solutions aside from an additional type, which seems worse.

All this to say, I don't think this concern is blocking, especially since the previous interface exposed was only 4 methods. Just wanted to share this feeling.